src/modules/ekf2/EKF/gps_control.cpp

```diff
@@ -81,12 +83,34 @@ void Ekf::controlGpsFusion()
 		const float pos_var = sq(pos_noise);
 		const Vector2f pos_obs_var(pos_var, pos_var);
 		updateHorizontalPositionAidSrcStatus(gps_sample.time_us,
-						     gps_sample.pos,                             // observation
+						     position,                                   // observation
 						     pos_obs_var,                                // observation variance
 						     math::max(_params.gps_pos_innov_gate, 1.f), // innovation gate
 						     _aid_src_gnss_pos);
 		_aid_src_gnss_pos.fusion_enabled = (_params.gnss_ctrl & GnssCtrl::HPOS);
 
+		// if GPS is otherwise ready to go, but yaw_align is blocked by EV give mag a chance to start
+		if (_control_status.flags.tilt_align && _NED_origin_initialised
+		    && gps_checks_passing && !gps_checks_failing) {
+
+			if (!_control_status.flags.yaw_align) {
```

Comment:
@bresch FYI in testing I also had the yaw estimator able to yaw_align here (eg transitioning from indoor -> outdoor), but I wanted to untangle more of mag control (https://github.com/PX4/PX4-Autopilot/pull/20470) before pursuing that upstream.